docs/forms.md

```diff
@@ -0,0 +1,143 @@
+# AcroForms in PDFKit
+
+AcroForms are interactive features of the PDF format, and they make it possible
+to include things like text fields, buttons and actions. To include AcroForms
+you must call the document `initAcroForm()` method.
+
+* `initAcroform()` - Must be called when using AcroForms
+
+## AcroForm Methods
+
+AcroForm elements are _Widget Annotations_ and are added using the
+`widgetAnnotation` method. Additional methods listed below are shortcut methods
+that call the `widgetAnnotation` method. The list of the available _Widget
+Annotation_ document methods is:
+
+* `widgetAnnotation( name, x, y, width, height, options)`
+* `formText( name, x, y, width, height, options)`
+* `formPushButton( name, x, y, width, height, name, options)`
+* `formRadioButton( name, x, y, width, height, options)`
+* `formNoToggleToOffButton( name, x, y, width, height, options)`
+* `formChoice( name, x, y, width, height, options)`
+
+### Options Parameter
+
+Some Widget Annotations have a `color` option that can be specified. You can use
+an array of RGB values, a hex color, or a named CSS color value for that option.
+
+* `backgroundColor` - button background color
+* `borderColor` - button border color
+
+Other `options` conveniences include:
+
+* `label` - set button text labels (will set <<MK <<CA (label)>> >>)
+* `align` - set to `left`, `center` or `right` for text within the
+  Widget Annotation
+* Field flags that will set bits in `Ff`
+  * `readyOnly`: 1,
+  * `required`: 2,
+  * `noExport`: 4,
+  * `multiline`: 0x1000,
+  * `password`: 0x2000,
+  * `toggleToOffButton`: 0x4000,
```

Comment:
From an API point of view, it might be nice to separate the field options from the field types. i.e. instead of choosing a field type by setting a boolean option, we could have a string "type" field for those, and leave the booleans for actual options. Or we could hide the field type options entirely from the API, and provide methods for them as you have for some of the below. The PDF format is often cryptic, so I don't think we necessarily need to expose it directly.

Reply:
`toggleToOffButton` is actually an option. I made a mistake in exposing this as a method.

Reply:
Please have a look when the next submit gets pushed